frontend/src/js/previous-queries/list/useDeleteProjectItemFolder.tsx

```diff
@@ -0,0 +1,86 @@
+import { StateT } from "app-types";
+import { useTranslation } from "react-i18next";
+import { useSelector, useDispatch } from "react-redux";
+
+import { useDatasetId } from "../../dataset/selectors";
+import { setMessage } from "../../snack-message/actions";
+
+import {
+  removeFolder,
+  useLoadFormConfigs,
+  useLoadQueries,
+  useUpdateFormConfig,
+  useUpdateQuery,
+} from "./actions";
+import type { FormConfigT, PreviousQueryT } from "./reducer";
+
+export const useDeleteProjectItemFolder = (
+  folder: string,
+  onSuccess?: () => void,
+) => {
+  const { t } = useTranslation();
+  const datasetId = useDatasetId();
+  const dispatch = useDispatch();
+  const { updateQuery } = useUpdateQuery();
+  const { updateFormConfig } = useUpdateFormConfig();
+  const { loadQueries } = useLoadQueries();
+  const { loadFormConfigs } = useLoadFormConfigs();
+
+  const queries = useSelector<StateT, PreviousQueryT[]>(
+    (state) => state.previousQueries.queries,
+  );
+  const formConfigs = useSelector<StateT, FormConfigT[]>(
+    (state) => state.previousQueries.formConfigs,
+  );
+
+  const localFolders = useSelector<StateT, string[]>(
+    (state) => state.previousQueries.localFolders,
+  );
+
+  return async () => {
+    if (!datasetId) return;
+
+    if (localFolders.includes(folder)) {
+      dispatch(removeFolder({ folderName: folder }));
+      if (onSuccess) {
+        onSuccess();
+      }
+      return;
+    }
+
+    try {
+      await Promise.all([
+        ...queries
+          .filter((query) => query.tags.includes(folder))
+          .map((query) => {
+            const nextTags = query.tags.filter((tag) => tag !== folder);
+
+            return updateQuery(
+              query.id,
+              { tags: nextTags },
+              t("previousQuery.retagError"),
+            );
+          }),
+        ...formConfigs
```

Comment:
It would be yes, especially because this is an n+1 problem. But it would drag a long tail of changes behind it. 

Until we get there, with http2 at least lots of parallel requests aren't as much of a problem from the browser-side anymore as they were back in http1, so we might get away with this for some time.